pkg/apis/autoscaling/v1alpha1/hvpa_types.go

```diff
@@ -172,21 +237,84 @@ const (
 	In Scaling = "in"
 )
 
+// LastError has detailed information of the error
+type LastError struct {
+	// Description of the error
+	Description string `json:"description,omitempty"`
+
+	// Time at which the error occurred
+	LastUpdateTime metav1.Time `json:"lastUpdateTime,omitempty"`
+
+	// LastOperation is the type of operation for which error occurred
+	LastOperation string `json:"lastOperation,omitempty"`
+}
+
 // HvpaStatus defines the observed state of Hvpa
 type HvpaStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-	// current information about the HPA.
-	// +optional
-	HpaStatus HpaStatus `json:"hpaStatus,omitempty" protobuf:"bytes,1,opt,name=hpaStatus"`
+	// Replicas is the number of replicas of the target resource.
+	Replicas *int32 `json:"replicas,omitempty"`
+	// HpaSelector is the string form of the label selector of HPA. This is required for HPA to work with scale subresource.
+	HpaSelector *string `json:"hpaSelector,omitempty"`
```

Comment:
`(H|h)paSelector` is confusing. This selector is supposed to select the `pods` of the `targetRef`. If this is changed then the annotation [here](https://github.com/gardener/hvpa-controller/pull/19/files#diff-617e45ba1eda37bfddd8607785d008c6R332) also needs to be changed.